packages/react-components/form-elements/src/Switch/Toggle.tsx

```diff
@@ -0,0 +1,90 @@
+import { CheckmarkIcon } from '@datacamp/waffles-icons';
+import tokens from '@datacamp/waffles-tokens';
+import { css } from '@emotion/react';
+
+import { SwitchProps } from './index';
+
+const toggleWrapperStyle = css({
+  backgroundClip: 'content-box',
+  backgroundColor: tokens.colors.blue,
+  borderRadius: 12,
+  boxSizing: 'border-box',
+  height: 24,
+  marginRight: 8,
+  padding: 3,
+  position: 'relative',
+  width: 34,
+});
+
+const toggleButtonleStyle = css({
+  alignItems: 'center',
+  backgroundColor: tokens.colors.white,
+  borderRadius: 6,
+  boxSizing: 'border-box',
+  display: 'flex',
+  height: 12,
+  justifyContent: 'center',
+  left: 6,
+  position: 'absolute',
+  top: 6,
+  transition: 'all 75ms linear',
+  width: 12,
+  zIndex: 1,
+});
+
+type ToggleProps = {
+  hasLabel: boolean;
+  isFocusVisible: boolean;
+} & Pick<SwitchProps, 'appearance' | 'checked'>;
+
+function Toggle({
+  appearance,
+  checked,
+  hasLabel,
+  isFocusVisible,
+}: ToggleProps): JSX.Element {
+  function toggleBackgroundColor(): string {
+    if (!checked) {
+      if (appearance === 'inverted') {
+        return tokens.colors.white;
+      }
+      return tokens.colors.grey400;
+    }
+    return tokens.colors.blue;
+  }
+
+  return (
+    <div
+      css={css(
+        toggleWrapperStyle,
+        {
+          backgroundColor: toggleBackgroundColor(),
+          marginRight: hasLabel ? 8 : 0,
+        },
+        isFocusVisible && {
+          boxShadow: `0 0 0 2px ${tokens.colors.blueDark}`,
+        },
+      )}
+    >
+      <div
+        css={css(toggleButtonleStyle, {
+          backgroundColor:
+            !checked && appearance === 'inverted'
+              ? tokens.colors.grey400
+              : tokens.colors.white,
+          left: checked ? 16 : 6,
```

Comment:
`transform: translateX(10px)` would be more performant because it's a compositor-only animation
https://developers.google.com/web/fundamentals/performance/rendering/stick-to-compositor-only-properties-and-manage-layer-count

Reply:
Background color also needs to paint to animate of course, but animating left would also need to layout.

Reply:
Maybe `will-change` could be added too.

Reply:
Good point, probably it even makes more sense to use **translateX** anyway.

Reply:
As for `will-change` I think it could be an overkill.